odoo/addons/base/models/res_lang.py

```diff
@@ -279,13 +281,20 @@ def toggle_active(self):
             mods = self.env['ir.module.module'].search([('state', '=', 'installed')])
             mods._update_translations(active_lang)
 
-    @api.model_create_multi
-    def create(self, vals_list):
-        self.clear_caches()
-        for vals in vals_list:
-            if not vals.get('url_code'):
-                vals['url_code'] = vals.get('iso_code') or vals['code']
-        return super(Lang, self).create(vals_list)
+        # If we activate a lang, set it's url_code to the shortest version if possible
+        if self.active and len(self.url_code) > 3:   # mya is the shortest
+            short = self.code.split('_')[0]
+            others = self.with_context(active_test=False).search([('active', '=', False), ('url_code', '=', short)])
+            if others and others != self:
+                others.url_code = others.code
+            if not self.search([('url_code', '=', short)]):
+                self.url_code = short
+
+    @api.depends('iso_code', 'code')
+    def _default_url_code(self):
+        for lang in self:
+            newlang = lang.iso_code or lang.code
+            lang.url_code = (newlang != 'my') and newlang or 'mya'
```

Comment:
Why is this `mya` needed?
We already make sure of that with the XML data: https://github.com/odoo/odoo/blame/07ff047b9dea098135284fedab3f34bf0e31d11b/odoo/addons/base/data/res_lang_data.xml#L7-L9